packages/driver/src/cypress/resolvers.ts

```diff
@@ -14,8 +14,17 @@ export function resolveWindowReference (this: typeof $Cypress, currentWindow: Wi
 
   const getTargetValue = () => {
     const targetValue = accessedObject[accessedProp]
+    const accessingDocument = dom.isDocument(accessedObject)
+    const hasLocation = dom.isWindow(accessedObject) || accessedObject
```

Comment:
this is such a weird condition. is it a boolean? Probably due to `dom.isWindow`. Is it an accessed object? sure, why not. and then, if it has location and it is accessing the document, we set accessed object ... to the current window.

Reply:
yes, it is weird :disappointed: i gave everything boolean-esque names to try and help to make the weirdness more easy to parse

Reply:
oh i see there's a bug here too, it should be accessingDocument, not accessedObject